packages/catalog-server/src/lib/firestore/firestore-repository.ts

```diff
@@ -364,13 +370,34 @@ export class FirestoreRepository implements Repository {
    */
   async getPackageVersion(
     packageName: string,
-    version: string
-  ): Promise<Omit<PackageVersion, 'customElements' | 'problems'> | undefined> {
-    const versionDoc = await this.getPackageVersionRef(
-      packageName,
-      version
-    ).get();
-    return versionDoc.data();
+    versionOrTag: string
+  ): Promise<PackageVersion | undefined> {
+    const versionNumber = clean(versionOrTag);
+
+    if (versionNumber !== null) {
+      // If version is valid semver we can build a document reference.
+      const versionRef = this.getPackageVersionRef(packageName, versionNumber);
+      const versionDoc = await versionRef.get();
+      return versionDoc.data();
+    } else {
+      // If version is not a valid semver it may be a dist-tag
+
+      // First, filter out semver ranges, since npm doesn't allow semver
```

Comment:
Are we not going to support semver ranges in our API?

Reply:
I wasn't planning on it, I'm not sure what the use case is. And if we did, we'd need to figure out how to query (starting with a lexicographic range and filtering?) and in which gql queries it would make sense to enable.